### DISCUSSION TITLE
Apparatus levels playable in Principia?

### DISCUSSION THREAD

**Discussion by adrian-maggio** [Bithack/principia#31] [General]
I apologize if this has been asked/answered elsewhere.
Will Principia support loading old Apparatus level saves? If not, are there any plans to add this feature?
Maybe there is a possibility for someone to create a tool to convert Apparatus levels to Principia levels?

**Comment by rollerozxa** [comment]
There are a lot of things different between Apparatus and Principia, including the basic set of objects that they share. Theoretically one could make a converter, the Apparatus file format has been fully reverse engineered IIRC and the Principia level format can be found just by looking at the source code. But it would most likely generate broken levels that would need a lot of work fixing, at that point you might as well recreate them in Principia manually.
